lib/core/utils/preload-cssom.js

```diff
@@ -0,0 +1,231 @@
+	Array.from(root.styleSheets).forEach(sheet => {
+		// ignore disabled sheets
+		if (sheet.disabled) {
+			return;
+		}
+		// attempt to retrieve cssRules, or for external sheets make a XMLHttpRequest
+		try {
+			// accessing .cssRules throws for external (cross-domain) sheets, which is handled in the catch
+			const cssRules = sheet.cssRules;
+			// read all css rules in the sheet
+			const rules = Array.from(cssRules);
+			// filter rules to assess if any external rules by way of @import or link exists on a nested level
+			const externalRules = rules.filter(r => r.href);
+			// if no external rules then return an expanded version of
+			if (!externalRules.length) {
+				q.defer(resolve => resolve(getCssomSheet(sheet, false, shadowId)));
+				return;
+			}
+
+			// there are external rules, best to filter non-external rules and create a new stylesheet to avoid duplication
+			const nonExternalRules = rules.filter(rule => !rule.href);
+			// concat all cssText into a string for non external rules
+			const nonExternalRulesText = nonExternalRules
+				.reduce((out, rule) => {
+					out.push(rule.cssText);
+					return out;
+				}, [])
+				.join();
+
+			// create a dynamic sheet for non external rules
+			q.defer(resolve =>
+				resolve(
+					convertTextToStylesheetFn({
+						data: nonExternalRulesText,
+						shadowId,
+						isExternal: false
+					})
+				)
+			);
+
+			// for any external rule -> make an xhr and q the response
+			externalRules.forEach(rule => {
+				q.defer((resolve, reject) => {
+					getExternalStylesheet({ resolve, reject, url: rule.href });
+				});
+			});
+		} catch (e) {
+			// if no href, do not attempt to make an XHR
+			if (!sheet.href) {
+				return;
+			}
+			// external sheet -> make an xhr and q the response
+			q.defer((resolve, reject) => {
+				getExternalStylesheet({ resolve, reject, url: sheet.href });
+			});
+		}
+	}, []);
+	// return
+	return q;
+}
+
+/**
+ * Returns an array of objects with root(document)
+ * @param {Object} treeRoot the DOM tree to be inspected
+ * @return {Array<Object>} array of objects, which each object containing a root (document) and an optional shadowId
+ * @private
+ */
+function getAllRootsInTree(tree) {
+	/**
+	 * Clone of axe.commons.dom.getRootNode
+	 * Note:
+	 * unable to call - axe.commons.dom.getRootNode(node.actualNode) while running tests, in any part of the code in lib/core,
+	 * as commons is not loaded when running core tests and vice versa, in test suite due to grunt concat linearity and seperation of files
+	 * so taking a copy of the function here for now
+	 * TODO:
+	 * cleanup later, once build is re-engineered
+	 * @method getRootNode
+	 * @param {Element} node
+	 * @returns {DocumentFragment|Document}
+	 * @private
+	 */
+	function getRootNode(node) {
+		var doc = (node.getRootNode && node.getRootNode()) || document; // this is for backwards compatibility
```

Comment:
Backwards compatibility with what?

Reply:
Cloned/ Copy pasted method. 
Going to try and fix the runner template to have axe.commons instead.